Deprecate reconfigurable flag

tsyesika opened the pull request:
This deprecates the `--reconfigurable` flag in favour of having it always be in a multi-process re-configurable mode. The use of the flag will now raise a deprecation warning.

One other thing to note is as of this PR the ingress drop monitor support is no longer available as documented in #902.

wingo commented on src/program/lwaftr/bench/bench.lua:
```diff
@@ -31,7 +31,9 @@ function parse_args(args)
    function handlers.h() show_usage(0) end
-   function handlers.reconfigurable() opts.reconfigurable = true end
+   function handlers.reconfigurable()
+      io.stderr:write("Warning: --reconfigurable flag has been deprecated.\n")
```
Let's remove this one entirely, as this is an internal script that's not user-facing.

wingo commented on src/program/lwaftr/run/run.lua:
```diff
@@ -102,7 +102,9 @@ function parse_args(args)
    end
-   function handlers.reconfigurable() opts.reconfigurable = true end
+   function handlers.reconfigurable()
+      io.stderr:write("Warning: --reconfigurable flag has been deprecated.\n")
```
Can we adapt this note to somehow assure the user that --reconfigurable is always enabled so they can safely remove the flag from their scripts?  This note says that it's deprecated but not what action they need to take.

wingo reviewed (approved):
excellent, thank you!

Commit:
Deprecate reconfigurable flag

This deprecates the --reconfigurable flag from all lwaftr apps. The
reconfigurable / multiprocess mode is always enabled. The tests have
been simplied to reflect this. A deprecation warning is shown if
--reconfigurable is used.

As of this change, ingress drop monitor is no longer supported. An
issue tracking the support is #902. A warning is shown if it is used.

Commit:
Fix few nits with depreaction of --reconfigurable

This removes the --reconfigurable flag in the bench command entirely
as it's internal facing. It also improves the message in the
deprecation warning in the run command.
